nixos/modules/services/monitoring/telegraf.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.telegraf;
+
+  configFile = pkgs.runCommand "config.toml" {
+    buildInputs = [ pkgs.remarshal ];
+  } ''
+    remarshal -if json -of toml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.extraConfig)} \
+      > $out
+  '';
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.telegraf = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the telegraf server";
+        type = types.bool;
+      };
+
+      package = mkOption {
```

Comment:
I often have packages that are outdated on nixos. Being able to quickly override the package from my configuration.nix allows me to quickly test if a new version will need some changes to the nix definition or not.

TBH, I can remove this either way. I included it mostly because it seems to be there in most other nix services. I've often found it useful, so I kind of just always put it there.